ang/test/civiawards/award-creation/directives/award-custom-field-sets.directive.spec.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-env jasmine */
+(function (_, $) {
+  describe('civiawardCustomFieldSets', () => {
+    var civiawardCustomFieldSetsDirective, $compile, $rootScope, $scope;
+
+    beforeEach(module('civiawards.templates', 'civiawards'));
+
+    beforeEach(inject((_$compile_, _$rootScope_) => {
+      $compile = _$compile_;
+      $rootScope = _$rootScope_;
+    }));
+
+    describe('basic tests', () => {
+      beforeEach(() => {
+        initDirective();
+      });
+
+      it('shows the custom field set UI', () => {
+        expect(CRM.loadForm).toHaveBeenCalledWith('/civicrm/award/customfield?entityId=5', {
+          target: jasmine.any(Object)
+        });
+      });
+    });
+
+    describe('when saving the award', () => {
+      var promise, customFieldSetsTabCallbackFn;
+
+      beforeEach(() => {
+        initDirective();
+
+        var customFieldSetsTabObj = _.find($scope.tabs, function (tabObj) {
+          return tabObj.name === 'customFieldSets';
+        });
+
+        spyOn($.fn, 'trigger').and.callThrough();
+        spyOn($.fn, 'unblock').and.callThrough();
```

Comment:
when overriding functions like this we need to restore them in an `afterEach` otherwise we risk breaking other tests.